skema/program_analysis/CAST/python/ts2cast.py

```diff
@@ -0,0 +1,203 @@
+import json
+import os.path
+from pathlib import Path
+from typing import Any, Dict, List, Union
+
+from tree_sitter import Language, Parser, Node
+
+from skema.program_analysis.CAST2FN.cast import CAST
+from skema.program_analysis.CAST2FN.model.cast import (
+    Module,
+    SourceRef,
+    Assignment,
+    LiteralValue,
+    Var,
+    VarType,
+    Name,
+    Operator,
+    AstNode,
+    SourceCodeDataType,
+    ModelImport,
+    FunctionDef,
+    Loop,
+    Call,
+    ModelReturn,
+    ModelIf,
+    RecordDef,
+    Attribute,
+)
+
+from skema.program_analysis.CAST.python.node_helper import (
+    NodeHelper,
+)
+from skema.program_analysis.CAST.fortran.util import generate_dummy_source_refs
+from skema.program_analysis.CAST.fortran.variable_context import VariableContext
+
+from skema.program_analysis.tree_sitter_parsers.build_parsers import INSTALLED_LANGUAGES_FILEPATH
+
+
+PYTHON_VERSION = "3.10"
+
+class TS2CAST(object):
+    def __init__(self, source_file_path: str, from_file = True):
+        # from_file flag is used for testing purposes, when we don't have actual files
+        if from_file:
+            self.path = Path(source_file_path)
+            self.source_file_name = self.path.name
+            
+            # Python doesn't have a preprocessing step like fortran
+            self.source = self.path.read_text()
+        else:
+            self.path = "None"
+            self.source_file_name = "Temp"
+            self.source = source_file_path
+
+        # Run tree-sitter preprocessor output to generate parse tree
+        parser = Parser()
+        parser.set_language(
+            Language(
+                INSTALLED_LANGUAGES_FILEPATH,
+                "python"
+            )
+        )
+
+        # Tree walking structures
+        self.variable_context = VariableContext()
+        self.node_helper = NodeHelper(self.source, self.source_file_name)
+
+        self.tree = parser.parse(bytes(self.source, "utf8"))
+
+        self.out_cast = self.generate_cast()
+
+    def generate_cast(self) -> List[CAST]:
+        '''Interface for generating CAST.'''
+        module = self.run(self.tree.root_node)
+        return CAST([generate_dummy_source_refs(module)], "Python") 
+        
+    def run(self, root) -> List[Module]:
+        # In python there's generally only one module at the top level
```

Comment:
This is my understanding as well. Of our current supported languages, I believe only Fortran can have multiple modules in the same source file.